### ISSUE TITLE
Backpropagation Algorithm

### ISSUE THREAD

**Issue by Enzogall** [MathEpitech2025/my_torch#4]
Implement backpropagation for supervised learning:

* Compute gradients for each layer
* Implement gradient descent
* Calculate loss/cost function
* Update weights and biases
